pkg/storage/stores/shipper/indexshipper/downloads/util.go

```diff
@@ -22,6 +24,9 @@ func (m *mtxWithReadiness) markReady() {
 }
 
 func (m *mtxWithReadiness) awaitReady(ctx context.Context) error {
+	ctx, cancel := context.WithTimeoutCause(ctx, 30*time.Second, errors.New("exceeded 30 seconds in awaitReady"))
```

Comment:
The 30s timeout seems somewhat arbitrary, but it proved to be a reasonable value.